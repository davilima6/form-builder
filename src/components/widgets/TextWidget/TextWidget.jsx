// @flow
import * as React from 'react';
import styled, { css } from 'styled-components';
import { normalize } from '../../../utils';
import {
  Label, StyledError, StyledInput, StyledLabel, WidgetWrapper,
} from '../..';

const { useCallback } = React;

type Props = {
  id: string,
  label: string,
  value?: number,
  errors?: Array<string>,
  onChangeField: Function,
};

const TextWidget = ({
  id, label, value, errors, onChangeField,
}: Props) => {
  const fieldId = `field-${id}`;
  const onChange = useCallback(({ target }: SyntheticEvent<>) => onChangeField(id, target.value), [
    id,
    onChangeField,
  ]);

  return (
    <WidgetWrapper error={errors.length}>
      <Label htmlFor={fieldId}>
        <StyledLabel>{label}</StyledLabel>
        <StyledInput type="text" id={fieldId} name={id} value={value} onChange={onChange} />
      </Label>
      {errors.map(message => (
        <StyledError key={normalize(message)}>{message}</StyledError>
      ))}
    </WidgetWrapper>
  );
};

TextWidget.defaultProps = {
  value: '',
  errors: [],
};

export default TextWidget;
