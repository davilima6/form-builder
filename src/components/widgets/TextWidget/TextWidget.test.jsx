import React from 'react';
import { cleanup, render } from 'react-testing-library';
import 'jest-dom/extend-expect';
// import 'jest-styled-components';
import TextWidget from './TextWidget';

describe('TextWidget', () => {
  let component;

  beforeEach(() => {
    const props = {
      label: 'First Name',
      type: 'text',
      data: 'John',
      errors: [],
      onChange: () => {},
    };

    cleanup();
    component = render(<TextWidget {...props} />);
  });

  test('renders without crashing', () => expect(component).toBeDefined());

  test.skip('matches snapshot', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
    expect(asFragment()).toHaveStyleRule('background', '#fff');
  });
});
