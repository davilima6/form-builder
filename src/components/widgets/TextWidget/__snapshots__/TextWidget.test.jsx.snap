// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWidget matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fjdhpX fQffyr"
  >
    <label
      class="sc-gZMcBi giNnHm"
      for="field-undefined"
    >
      <div
        class="sc-jTzLTM jXxOTJ"
      >
        First Name
      </div>
      <input
        class="sc-VigVT dQqCzP"
        id="field-undefined"
        type="text"
        value=""
      />
    </label>
  </div>
</DocumentFragment>
`;
